fakeserver/routes.go

```diff
@@ -257,6 +261,35 @@ func postV1AssignmentOverride(r *http.Request) error {
 	return nil
 }
 
+func postV2AssignmentOverride(r *http.Request) error {
+	var assignments []v1Assignment
+	contentType := r.Header.Get("content-type")
+	switch {
+	case strings.HasPrefix(contentType, "application/json"):
+		requestBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return err
+		}
+		var assignmentMap map[string][]v1Assignment
+		err = json.Unmarshal(requestBytes, &assignmentMap)
+		if err != nil {
+			return err
+		}
+		assignments = assignmentMap["assignments"]
+	default:
+		return fmt.Errorf("got unexpected content type %s", contentType)
+	}
+	storedAssignments, err := fakeassignments.Read()
+	for _, assignment := range assignments {
+		(*storedAssignments)[assignment.SplitName] = assignment.Variant
+	}
```

Comment:
this is the other new bit compared to the previous singular version. before this was a single assignment, now it's a loop ➰